pkg/blobinfocache/internal/prioritize/prioritize.go

```diff
@@ -77,34 +82,58 @@ func (css *candidateSortState) Swap(i, j int) {
 	css.cs[i], css.cs[j] = css.cs[j], css.cs[i]
 }
 
-// destructivelyPrioritizeReplacementCandidatesWithMax is destructivelyPrioritizeReplacementCandidates with a parameter for the
-// number of entries to limit, only to make testing simpler.
-func destructivelyPrioritizeReplacementCandidatesWithMax(cs []CandidateWithTime, primaryDigest, uncompressedDigest digest.Digest, maxCandidates int) []blobinfocache.BICReplacementCandidate2 {
+// destructivelyPrioritizeReplacementCandidatesWithMax is destructivelyPrioritizeReplacementCandidates with parameters for the
+// number of entries to limit for known and unknown location separately, only to make testing simpler.
+func destructivelyPrioritizeReplacementCandidatesWithMax(cs []CandidateWithTime, primaryDigest, uncompressedDigest digest.Digest, maxCandidates int, unknownLocationLimit int) []blobinfocache.BICReplacementCandidate2 {
+	// split unknown candidates and known candidates
+	// and limit them seperately.
+	var knownLocationCandidates []CandidateWithTime
+	var unknownLocationCandidates []CandidateWithTime
+
+	for _, candidate := range cs {
+		if candidate.Candidate.UnknownLocation {
+			unknownLocationCandidates = append(unknownLocationCandidates, candidate)
```

Comment:
Non-blocking: We can completely filter out `primaryDigest` entries with unknown locations, because `dockerImageDestination` is going to try that digest anyway, and discard them from the candidate list. So doing that would allow us to add one more _relevant_ candidate.

Reply:
Leaving this for now, will catch this low-hanging fruit in a new PR ( since its a non-blocker )